cellxgene_schema_cli/cellxgene_schema/validate.py

```diff
@@ -644,18 +644,30 @@ def _validate_dataframe(self, df_name: str):
                     f"Features SHOULD NOT be filtered from expression matrix."
                 )
 
-        # Check for columns that have a category defined 0 times (obs only)
-        if df_name == "obs":
-            for column in df.columns:
-                col = df[column]
-                if col.dtype != "category":
-                    continue
-                for category in col.dtype.categories:
-                    if category not in col.values:
-                        self.warnings.append(
-                            f"Column '{column}' in dataframe '{df_name}' contains a category '{category}' with "
-                            f"zero observations. These categories will be removed when `--add-labels` flag is present."
-                        )
+        for column_name in df.columns:
+            column = df[column_name]
+            if column.dtype != "category":
```

Comment:
curious, why check for != category? is it because that dtype would already validate that they do not have mixed values?

Reply:
nvm--I see now that it has its own type-check

Reply:
A regular ndarray doesn't require a specifing a dtype so it could have mixed values. I'm assuming that Catagorical restricts to the specified catagories which are typed, but I haven't confirmed that.